### PR TITLE
Modernize tests code

### PR DESCRIPTION
**Pull request by xPaw**
Applied Visual studio suggestions.

### CODE REVIEW

File: SteamKit2/Tests/CallbackManagerFacts.cs [hunk 1]
```diff
@@ -26,13 +26,13 @@ public void PostedCallbackTriggersAction()
             var callback = new CallbackForTest { UniqueID = Guid.NewGuid() };
 
             var didCall = false;
-            Action<CallbackForTest> action = delegate(CallbackForTest cb)
+            void action( CallbackForTest cb )
```
Comment (yaakov-h) [inline SteamKit2/Tests/CallbackManagerFacts.cs:29]: this change seem pointless - converting the delegate to a local function, only to have to cast it back to the delegate a couple lines down

File: SteamKit2/Tests/CallbackManagerFacts.cs [hunk 2]
```diff
@@ -46,15 +46,15 @@ public void PostedCallbackTriggersAction_CatchAll()
             var callback = new CallbackForTest { UniqueID = Guid.NewGuid() };
 
             var didCall = false;
-            Action<CallbackMsg> action = delegate(CallbackMsg cb)
+            void action( CallbackMsg cb )
```
Comment (yaakov-h) [inline SteamKit2/Tests/CallbackManagerFacts.cs:49]: and here

File: SteamKit2/Tests/CallbackManagerFacts.cs [hunk 3]
```diff
@@ -69,14 +69,14 @@ public void PostedCallbackTriggersActionForExplicitJobIDInvalid()
             var callback = new CallbackForTest { JobID = jobID, UniqueID = Guid.NewGuid() };
 
             var didCall = false;
-            Action<CallbackForTest> action = delegate(CallbackForTest cb)
+            void action( CallbackForTest cb )
```
Comment (yaakov-h) [inline SteamKit2/Tests/CallbackManagerFacts.cs:72]: etc.

File: SteamKit2/Tests/NetHelpersFacts.cs [hunk 4]
```diff
@@ -11,12 +11,12 @@ public class NetHelpersFacts
         public void GetMsgIPAddress()
         {
             Assert.Equal( 2130706433u, NetHelpers.GetMsgIPAddress( IPAddress.Loopback ).v4 );
-            Assert.Equal( new byte[] {
+            Assert.Equal( [
```
Comment (yaakov-h) [inline SteamKit2/Tests/NetHelpersFacts.cs:14]: ... the target-typing picks this up as a byte array literal?

File: SteamKit2/Tests/StreamHelpersFacts.cs [hunk 5]
```diff
@@ -36,46 +36,44 @@ public void IsThreadSafe()
             }
         }
 
-        ConcurrentBag<Exception> threadExceptions = new ConcurrentBag<Exception>();
+        ConcurrentBag<Exception> threadExceptions = [];
```
Comment (yaakov-h) [inline SteamKit2/Tests/StreamHelpersFacts.cs:39]: I'm surprised the target typing works with this, I would have expected `= new();`

File: SteamKit2/Tests/StreamHelpersFacts.cs [hunk 6]
```diff
@@ -36,46 +36,44 @@ public void IsThreadSafe()
...
-                    default: throw new Exception("Unreachable");
-                }
+                    0 => s.ReadByte(),
+                    1 => s.ReadInt64(),
+                    2 => s.ReadUInt16(),
+                    3 => s.ReadInt32(),
+                    4 => s.ReadUInt32(),
+                    5 => s.ReadInt64(),
+                    6 => s.ReadUInt64(),
+                    _ => throw new Exception( "Unreachable" ),
```
Comment (yaakov-h) [inline SteamKit2/Tests/StreamHelpersFacts.cs:75]: Consider replacing with `UnreachableException`

File: SteamKit2/Tests/StreamHelpersFacts.cs (hunk 5 above)
Comment (xPaw) [reply SteamKit2/Tests/StreamHelpersFacts.cs:39]: It works and VS automatically converts to this.

Comment (yaakov-h) [reply SteamKit2/Tests/StreamHelpersFacts.cs:39]: interesting. both produce the same IL anyway so seems fine, just surprising.

File: SteamKit2/Tests/NetHelpersFacts.cs (hunk 4 above)
Comment (xPaw) [reply SteamKit2/Tests/NetHelpersFacts.cs:14]: Yes.

**Review by yaakov-h — APPROVED** [review]

**Pull request merged**